pkg/controller/nodeipam/ipam/cloud_cidr_allocator_test.go

```diff
@@ -83,3 +84,99 @@ func TestNodeUpdateRetryTimeout(t *testing.T) {
 		})
 	}
 }
+
+func TestNeedPodCIDRsUpdate(t *testing.T) {
+	for _, tc := range []struct {
+		desc         string
+		cidrs        []string
+		nodePodCIDR  string
+		nodePodCIDRs []string
+		want         bool
+		wantErr      bool
+	}{
+		{
+			desc:         "want error - invalid cidr",
+			cidrs:        []string{"10.10.10.0/24"},
+			nodePodCIDR:  "10.10..0/24",
+			nodePodCIDRs: []string{"10.10..0/24"},
+			wantErr:      true,
+		},
+		{
+			desc:         "want false - matching v4 only cidr",
+			cidrs:        []string{"10.10.10.0/24"},
+			nodePodCIDR:  "10.10.10.0/24",
+			nodePodCIDRs: []string{"10.10.10.0/24"},
+			want:         false,
+		},
+		{
+			desc:  "want false - nil node.Spec.PodCIDR",
+			cidrs: []string{"10.10.10.0/24"},
+			want:  true,
+		},
+		{
+			desc:         "want true - non matching v4 only cidr",
+			cidrs:        []string{"10.10.10.0/24"},
+			nodePodCIDR:  "10.10.11.0/24",
+			nodePodCIDRs: []string{"10.10.11.0/24"},
+			want:         true,
+		},
+		{
+			desc:         "want false - matching v4 and v6 cidrs",
+			cidrs:        []string{"10.10.10.0/24", "2001:db8::/64"},
+			nodePodCIDR:  "10.10.10.0/24",
+			nodePodCIDRs: []string{"10.10.10.0/24", "2001:db8::/64"},
+			want:         false,
+		},
+		{
+			desc:         "want false - matching v4 and v6 cidrs, different strings but same CIDRs",
+			cidrs:        []string{"10.10.10.0/24", "2001:db8::/64"},
+			nodePodCIDR:  "10.10.10.0/24",
+			nodePodCIDRs: []string{"10.10.10.0/24", "2001:db8:0::/64"},
+			want:         false,
+		},
+		{
+			desc:         "want true - matching v4 and non matching v6 cidrs",
+			cidrs:        []string{"10.10.10.0/24", "2001:db8::/64"},
+			nodePodCIDR:  "10.10.10.0/24",
+			nodePodCIDRs: []string{"10.10.10.0/24", "2001:dba::/64"},
+			want:         true,
+		},
+		{
+			desc:  "want true - nil node.Spec.PodCIDRs",
+			cidrs: []string{"10.10.10.0/24", "2001:db8::/64"},
+			want:  true,
+		},
+		{
+			desc:         "want true - matching v6 and non matching v4 cidrs",
+			cidrs:        []string{"10.10.10.0/24", "2001:db8::/64"},
+			nodePodCIDR:  "10.10.1.0/24",
+			nodePodCIDRs: []string{"10.10.1.0/24", "2001:db8::/64"},
+			want:         true,
+		},
+		{
+			desc:         "want true - missing v6",
+			cidrs:        []string{"10.10.10.0/24", "2001:db8::/64"},
+			nodePodCIDR:  "10.10.10.0/24",
+			nodePodCIDRs: []string{"10.10.10.0/24"},
+			want:         true,
+		},
+	} {
+		var node v1.Node
+		node.Spec.PodCIDR = tc.nodePodCIDR
+		node.Spec.PodCIDRs = tc.nodePodCIDRs
+		netCIDRs, err := netutils.ParseCIDRs(tc.cidrs)
+		if err != nil {
+			t.Errorf("failed to parse %v as CIDRs: %v", tc.cidrs, err)
+		}
+
+		t.Run(tc.desc, func(t *testing.T) {
+			got, err := needPodCIDRsUpdate(&node, netCIDRs)
+			if tc.wantErr == (err == nil) {
+				t.Errorf("err: %v, wantErr: %v", err, tc.wantErr)
+			}
```

Comment:
if there is an error, don't check got? (if .. else if)

Reply:
Done. 
Added ' if err == nil && got != tc.want { ' to check got only when err == nil.